### PR TITLE
Implements service monitor resource

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Closes https://github.com/giantswarm/giantswarm/issues/8119

### CODE REVIEW

File: service/controller/resource/servicemonitor/create.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	services, err := toServiceMonitors(obj)
```
Comment (JosephSalisbury) [inline service/controller/resource/servicemonitor/create.go]: i reckon this should be `serviceMonitors` - `services` is confusing

Comment (TheoBrigitte) [reply service/controller/resource/servicemonitor/create.go]: fixed

File: service/controller/resource/servicemonitor/create.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	serviceMonitors, err := toServiceMonitors(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	for _, service := range serviceMonitors {
```
Comment (JosephSalisbury) [inline service/controller/resource/servicemonitor/create.go]:
```suggestion
	for _, serviceMonitor := range serviceMonitors {
```

File: service/controller/resource/servicemonitor/create.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+		_, err = r.prometheusClient.MonitoringV1().ServiceMonitors(service.GetNamespace()).Create(service)
```
Comment (JosephSalisbury) [inline service/controller/resource/servicemonitor/create.go]:
```suggestion
		_, err = r.prometheusClient.MonitoringV1().ServiceMonitors(service.GetNamespace()).Create(serviceMonitor)
```

File: service/controller/resource/servicemonitor/delete.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
+	services, err := toServiceMonitors(obj)
```
Comment (JosephSalisbury) [inline service/controller/resource/servicemonitor/delete.go]:
```suggestion
	serviceMonitors, err := toServiceMonitors(obj)
```

File: service/controller/resource/servicemonitor/delete.go [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	for _, service := range services {
```
Comment (JosephSalisbury) [inline service/controller/resource/servicemonitor/delete.go]:
```suggestion
	for _, serviceMonitor := range serviceMonitors {
```

File: service/controller/resource/servicemonitor/delete.go [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+		err := r.prometheusClient.MonitoringV1().ServiceMonitors(service.GetNamespace()).Delete(service.GetName(), &metav1.DeleteOptions{})
```
Comment (JosephSalisbury) [inline service/controller/resource/servicemonitor/delete.go]:
```suggestion
		err := r.prometheusClient.MonitoringV1().ServiceMonitors(service.GetNamespace()).Delete(serviceMonitor.GetName(), &metav1.DeleteOptions{})
```

**Pull request merged**